contracts/implementations/tableland/VehicleTable.sol

```diff
@@ -0,0 +1,107 @@
+//SPDX-License-Identifier: BUSL-1.1
+/* solhint-disable */
+pragma solidity ^0.8.13;
+
+import "../../shared/Roles.sol";
+import "../../libraries/tableland/VehicleTableStorage.sol";
+import "../../libraries/nodes/ManufacturerStorage.sol";
+
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "@tableland/evm/contracts/utils/TablelandDeployments.sol";
+import "@tableland/evm/contracts/utils/SQLHelpers.sol";
+import "@solidstate/contracts/access/access_control/AccessControlInternal.sol";
+
+import "hardhat/console.sol";
+
+error InvalidManufacturerId(uint256 id);
+error TableAlreadyExists(uint256 tableId);
+
+/// TODO Documentation
+contract VehicleTable is AccessControlInternal {
+    event VehicleTableCreated(uint256 manufacturerId, uint256 tableId);
+
+    /// TODO Documentation
+    function createVehicleTable(
+        address owner,
+        uint256 manufacturerId
+    ) external onlyRole(ADMIN_ROLE) {
+        VehicleTableStorage.Storage storage vs = VehicleTableStorage
+            .getStorage();
+
+        string memory prefix = ManufacturerStorage
+            .getStorage()
+            .nodeIdToManufacturerName[manufacturerId];
+
+        if (bytes(prefix).length == 0) {
+            revert InvalidManufacturerId(manufacturerId);
+        }
+        if (vs.tables[manufacturerId] != 0) {
+            revert TableAlreadyExists(vs.tables[manufacturerId]);
```

Comment:
Maybe the argument here should just be the `manufacturerId`? The client can look up the id if he wants.